src/library/export/coverartcopyworker.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <QObject>
+#include <QScopedPointer>
+#include <QString>
+#include <QThread>
+#include <future>
+
+#include "util/fileinfo.h"
+#include "util/imagefiledata.h"
+
+// This is a QThread class for copying the cover art.
+
+class CoverArtCopyWorker : public QThread {
+    Q_OBJECT
+  public:
+    enum class OverwriteAnswer {
+        Overwrite,
+        Cancel = -1,
+    };
+
+    CoverArtCopyWorker(const ImageFileData& coverArtImage, const QString& coverArtAbsolutePath)
+            : m_coverArtImage(coverArtImage), m_coverArtAbsolutePath(coverArtAbsolutePath) {
+    }
+
+    virtual ~CoverArtCopyWorker(){};
+
+    void run() override;
+
+    bool isCoverUpdated() const {
+        return m_isCoverArtUpdated;
+    }
+
+  private:
+    void copyFile(const ImageFileData& m_coverArtImage,
+            const QString& m_coverArtAbsolutePath);
+
+    void askOverWrite(const QString& filename,
+            std::promise<CoverArtCopyWorker::OverwriteAnswer>* promise);
+
+    OverwriteAnswer makeOverwriteRequest(const QString& filename);
+
+    const ImageFileData& m_coverArtImage;
+    const QString m_coverArtAbsolutePath;
+    bool m_isCoverArtUpdated;
```

Comment:
If you plan to use this from two threads this needs to become an atomic

Reply:
Deleted 👍